### PR TITLE
[WIP] feat:优化地址位置信息的获取 (#112)

### PR DESCRIPTION
**Pull request by enjoyliu**
**Please provide issue(s) of this PR:**
Feat #112

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration
- [ ] Docs
- [ ] Performance and Scalability
- [ ] Naming
- [ ] HealthCheck
- [ ] Test and Release

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any user-facing changes. This may include API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: benchmark/stats/histogram.go [hunk 1]
```diff
@@ -106,7 +106,7 @@ func (h *Histogram) Print(w io.Writer) {
 	h.PrintWithUnit(w, 1)
 }
 
-// common printf
+// location printf
```
Comment (chuntaojun) [inline benchmark/stats/histogram.go:109]: ？这里修改的意义？

File: pkg/config/location_provider.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+)
+
+type LocationProviderConfigImpl struct {
```
Comment (chuntaojun) [inline pkg/config/location_provider.go:8]: 这里的配置实现保持和 polaris-java 那边一样吧

public interface LocationProviderConfig extends Verifier {

	String getTye();

	Map<String, Object> getOptions();

}

File: plugin/ratelimiter/common/ratelimit_slice_window.go [hunk 3]
```diff
@@ -15,7 +15,7 @@
  * specific language governing permissions and limitations under the License.
  */
 
-// Package common .
+// Package location .
```
Comment (chuntaojun) [inline plugin/ratelimiter/common/ratelimit_slice_window.go:18]: 这个修改回滚

**Review by chuntaojun — CHANGES_REQUESTED** [review]

File: benchmark/stats/histogram.go (hunk 1 above)
Comment (enjoyliu) [reply benchmark/stats/histogram.go]: refactor的时候带进来了，已回滚

File: plugin/location/location.go [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+func (p *Provider) GetLocation() (*model.Location, error) {
+	location := &model.Location{}
+
+	for _, plugin := range p.pluginChains {
+		tmp, err := plugin.GetLocation()
+		if err != nil {
+			log.GetBaseLogger().Errorf("get location from plugin %s error: %v", plugin.Name(), err)
+			continue
+		}
+		location = tmp
```
[the remark this answers is not in the document]
Comment (enjoyliu) [reply plugin/location/location.go:123]: ok,不过err为空的时候应该要保证这里location不为空吧

[the remark this answers is not in the document]
Comment (enjoyliu) [reply plugin/location/location.go:123]: 这里err为空的话是不是需要保证location不为空

File: pkg/config/location_provider.go (hunk 2 above)
Comment (enjoyliu) [reply pkg/config/location_provider.go:8]: ok,我参考一下

**Pull request merged**